### PR TITLE
[pallet-revive] rpc server add docker file

### PR DESCRIPTION
**Pull request by pgherveou**
Add a docker for pallet-revive eth-rpc

 Tested with 
```
sudo docker build . -t eth-rpc -f substrate/frame/revive/rpc/Dockerfile
sudo docker run --network="host" -e RUST_LOG="info,eth-rpc=debug" eth-rpc
```

### CODE REVIEW

File: substrate/frame/revive/rpc/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,23 @@
+FROM rust AS builder
+
+RUN apt-get update && \
+		DEBIAN_FRONTEND=noninteractive apt-get install -y \
+		protobuf-compiler
+
+WORKDIR /polkadot
+COPY . /polkadot
+RUN cargo build --locked --release -p pallet-revive-eth-rpc --bin eth-rpc
```
Comment (athei) [inline substrate/frame/revive/rpc/Dockerfile:9]:
```suggestion
RUN cargo build --locked --profile production -p pallet-revive-eth-rpc --bin eth-rpc
```

File: substrate/frame/revive/rpc/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+FROM docker.io/parity/base-bin:latest
+COPY --from=builder /polkadot/target/release/eth-rpc /usr/local/bin
```
Comment (athei) [inline substrate/frame/revive/rpc/Dockerfile:12]:
```suggestion
COPY --from=builder /polkadot/target/production/eth-rpc /usr/local/bin
```

**Review by xermicus — APPROVED** [review]

**Review by athei — APPROVED** [review]

**Comment by BulatSaif** [comment]
Why do we have a separate Dockerfile, and why isn’t eth-rpc part of the main polkadot-sdk? It’s not very convenient for end users to dive deep and research which binary to use for that purpose. In my opinion, this feature should be included in the main binary and enabled or disabled with a flag.

**Comment by BulatSaif** [comment]
As I understand it, eth-rpc is a wrapper around polkadot-rpc. For example, [substrate-api-sidecar](https://github.com/paritytech/substrate-api-sidecar) is also a wrapper around the RPC, but it’s in a separate repository and a separate binary.

**Comment by athei** [comment]
> and why isn’t eth-rpc part of the main polkadot-sdk binary

This is on purpose. We don't want to make the client more complicated. By adding it to the actual node every node implementation would be expected to expose additional endpoints adding an additional burden to alternative implementations.

One example is chopsticks which would need to add those endpoints in order to be able to be used with Ethereum clients.

**Pull request merged**